src/dapla_pseudo/v1/pseudo_commons.py

```diff
@@ -131,6 +131,13 @@ def get_content_type_from_file(file_handle: BinaryFileDecl) -> Mimetypes:
     return content_type
 
 
+@dataclass
+class HierarchicalDataFrame:
```

Comment:
Maybe this could be a type alias instead of a Dataclass? I.e. 

`## types.py
from typing import TypeAlias

HierarchicalDataFrame: TypeAlias = pl.DataFrame`

Reply:
@mallport nice suggestion, but `TypeAlias` will treat both types equivalently in type checking. So I couldnt't find a way to distinguish between `HierarchicalDataFrame` and `pl.DataFrame` in a `match` statement. 

I tried to `NewType` since static type checker will treat the new type as if it were a subclass of the original type. But that didn't work either - at least not in a `match` statement. So I'll have to keep using `@dataclass` unless you have an idea how to do it.

Reply:
Yeah.. it is kind of unfortunate, match cases don't match on the concrete type, but uses an `isinstance()` call under the hood, which checks for subtypes of the type in question. A solution could be to add a guard that also checks the concrete type:

```python
match Pseudonymize.dataset
    case HierarchicalDataset if type(Pseudonymize.dataset) == HierarchicalDataset
....
```

Up to you whether you want to do this, I'll resolve this for now :)